src/wp-admin/includes/plugin.php

```diff
@@ -2473,3 +2473,87 @@ function paused_plugins_notice() {
 		__( 'Go to the Plugins screen' )
 	);
 }
+
+/**
+ * Renders an admin notice in case some plugins deactivated.
+ *
+ * Displays an admin notice in case a plugin has been deactivated during an
+ * upgrade due to incompatibility with the current version of WordPress.
+ *
+ * @since 5.8.0
+ * @access private
+ *
+ * @global string $pagenow
+ */
+function deactivated_plugins_notice() {
+	if ( 'plugins.php' === $GLOBALS['pagenow'] ) {
+		return;
+	}
+
+	if ( ! current_user_can( 'activate_plugins' ) ) {
+		return;
+	}
+
+	$blog_deactivated_plugins = get_option( 'wp_deactivated_plugins' );
+	$site_deactivated_plugins = array();
+
+	if ( false === $blog_deactivated_plugins ) {
+		// Option not in database, add an empty array to avoid extra DB queries on subsequent loads.
+		update_option( 'wp_deactivated_plugins', array() );
```

Comment:
I'm wondering if `wp_deactivated_plugins` is not specific enough. I think the context is important here. Otherwise someone could just assume this is a list of any plugin that was once active and then deactivated. Maybe `wp_force_deactivated_plugins`? Or `wp_core_deactivated_plugins`? Naming is hard.